tests/rest/admin/test_room.py

```diff
@@ -175,15 +175,15 @@ def prepare(self, reactor, clock, hs):
         self.room_id = self.helper.create_room_as(
             self.other_user, tok=self.other_user_tok
         )
-        self.url = "/_synapse/admin/v1/rooms/%s/delete" % self.room_id
```

Comment:
I think we should test both the deprecated and new methods, as we don't want to accidentally break the old endpoint.

I think we can use `@parameterized_class` decorator to avoid duplicating code, so something like:

```python
@parameterized_class(
    ("method", "url_template"),
    [("POST", "/_synapse/admin/v1/rooms/%s/delete"), ("DELETE", "/_synapse/admin/v1/rooms/%s")]
)
class DeleteRoomTestCase(unittest.HomeserverTestCase):
    ...
``` 

See https://github.com/wolever/parameterized for some docs and https://github.com/matrix-org/synapse/blob/develop/tests/rest/media/v1/test_media_storage.py#L130 as an example.